libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/join/DefaultJoinRoom.kt

```diff
@@ -32,18 +33,30 @@ class DefaultJoinRoom @Inject constructor(
     private val analyticsService: AnalyticsService,
 ) : JoinRoom {
     override suspend fun invoke(
-        roomId: RoomId,
+        roomIdOrAlias: RoomIdOrAlias,
         serverNames: List<String>,
-        trigger: JoinedRoom.Trigger,
+        trigger: JoinedRoom.Trigger
     ): Result<Unit> {
-        return if (serverNames.isEmpty()) {
-            client.joinRoom(roomId)
-        } else {
-            client.joinRoomByIdOrAlias(roomId, serverNames)
-        }.onSuccess {
-            client.getRoom(roomId)?.use { room ->
-                analyticsService.capture(room.toAnalyticsJoinedRoom(trigger))
+        return when (roomIdOrAlias) {
+            is RoomIdOrAlias.Id -> {
+                if (serverNames.isEmpty()) {
+                    client.joinRoom(roomIdOrAlias.roomId)
+                } else {
+                    client.joinRoomByIdOrAlias(roomIdOrAlias, serverNames)
+                }
             }
+            is RoomIdOrAlias.Alias -> {
+                client.joinRoomByIdOrAlias(roomIdOrAlias, serverNames = emptyList())
+            }
+        }.onSuccess { roomSummary ->
+            client.captureJoinedRoomAnalytics(roomSummary, trigger)
+        }.map { }
+    }
+
+    private suspend fun MatrixClient.captureJoinedRoomAnalytics(roomSummary: RoomSummary?, trigger: JoinedRoom.Trigger) {
+        if (roomSummary == null) return
+        getRoom(roomSummary.roomId)?.use { room ->
+            analyticsService.capture(room.toAnalyticsJoinedRoom(trigger))
```

Comment:
It would be nice to avoide using `getRoom(...)?.use` to send this, since most of the info is already in the `RoomSummary`. We lack `isSpace` though 🫤 .

Reply:
Yea that's what I wanted to do at first.. we'll change later I guess!